doc/compiler/optimizer/GlobalOptimizationsSummary.md

```diff
@@ -0,0 +1,138 @@
+we can skip zero initialization completely.
+
+## Global Copy Propagation (Dead Store Elimination, Dead Structure Removal)
+These three analyses use use/def info to perform the traditional optimizations. The
+analysis is only done for locals (statics/fields omitted). Dead structure removal
+basically tries to find a region of code with no side-effects and having no definition
+that is used outside the structure; in such cases the (dead) region of code can be
+eliminated.
+
+## Redundant Monitor Elimination
+Redundant Monitor Elimination uses value numbering to perform nested monitor removal
+(e.g. syncenter on the same object twice followed by two syncexits). Use value
+number information and structure to perform lock coarsening (e.g. syncenter/syncexit
+on some object followed by another syncenter/syncexit on the same object). In this
+case the extent of code across which the lock is held is increased and one syncexit/syncenter
+can be removed. Deadlock avoidance is ensured by examining the code where the lock
+would be held after coarsening and making sure there are no calls or lock/unlock
+operations in that region of code. Sometimes additional syncenters/syncexits may be
+added on ‘cold’ paths to avoid some syncenters/syncexits on ‘hot’ paths.
+
+## Global Register Allocator (GRA)
+GRA is a tactical solution for achieving the effect of global register assignment,
+even though the underlying code generators have local register allocators. Basic idea
+is that global register dependencies are introduced in the IL trees; these dependencies
+(when evaluated by the code generators) ensure that a certain value is in a certain
+register on all control flow paths into a basic block. Stores/loads from memory are
+changed into stores/loads from registers in the IL (register numbers specify which
+physical register the value is placed into). Global register candidates are chosen
+by various optimizer passes like PRE, induction variable detection etc. and appropriate
+priorities are given to the candidates based on the number of references and live
+range (number of blocks). Advantages are that most of the code is platform neutral
+and it is relatively cheap in compile time in comparison to (expected cost of) a
+graph-coloring allocator.
+
+## Live Variables for GC (local and global)
+Live Variables for GC attempts to minimize the number of locals that need to be zero
+initialized upon method entry. Basic idea is to perform liveness analysis and find
+out if any locals are live at any [GC point](https://github.com/eclipse/omr/blob/81b79405da6c7c960e611a8b2b12fd5861543330/compiler/il/OMRNode.cpp#L2298)
+that is not dominated by (user defined) stores on all control flow paths. If a local
+is uninitialized at a GC point, explicit zero initialization must be inserted at the method entry.
+Note that this situation is fairly rare (only occurs in some uncommon cases involving jsr).
```

Comment:
"Involving JSR": do you mean JSR292? JSR by itself just refers to Java Specification Request.

Reply:
No, a long long time ago in a galaxy far far away, there used to be a bytecode known as `jsr` (jump to sub routine) that was used to implement throw/catch/finally when Java was first released. Because this code essentially caused some abnormal control flow, it could make it appear as though a variable was uninitialized on some code path. The `jsr` bytecode was removed a long time back. 

I believe the main situation in which such zero initializations may be needed in today's JIT compiler may be around some optimization creating such an appearance of uninitialized variables. 

I wonder if we should add the word "reference"  to "local" since the GC cares only about reference locals.